rpc/doc/README.md

```diff
@@ -0,0 +1,93 @@
+# RPC
+
+This document is meant to be a starting point/placeholder for a full-fledged RPC specification that allows interaction with the nodes.
+
+## Inspiration
+
+Pocket V0 has inspired a lot the first iteration but then we converged towards a spec-centric approach, where the boilerplate code (serialization, routing, etc) is derived from an [OpenAPI 3.0](../v1/openapi.yaml) specification.
+
+This approach will allow us to focus on the features and less on the boilerpate and ultimately to iterate more quickly as we discover the way ahead of us.
+
+## Code generation
+
+The current implementation uses code generation for ease of development.
+
+The source of truth is the the [OpenAPI3.0 yaml file](../v1/openapi.yaml).
+
+Anytime we make changes to the yaml file, we need to regenerate the boilerplate code by running
+
+```bash
+$ make generate_rpc_openapi
+```
+
+The compilation errors should guide towards the next steps.
+
+## Transports
+
+Currently, the API is in its simplest form. Basically a **REST API**.
+
+As the codebase matures, we'll consider other transports such as [**JSON RPC 2.0**](https://www.jsonrpc.org/specification) and [**GRPC**](https://grpc.io/).
+
+## Spec
+
+<!-- TODO (deblasis): add link when merged to `main` -->
+
+The Swagger Editor with preview is available here.
+
+This first iteration includes the bare minimum:
+
+### Node related
+
+- Node liveness check (**GET /v1/health**)
+- Node version check (**GET /v1/version**)
+
+These are pretty self-explanatory.
```

Comment:
:)

Reply:
Should I add a one liner each for safety?